test/e2e/cni/cni_test.go

```diff
@@ -0,0 +1,211 @@
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/amazon-vpc-cni-k8s/test/e2e/cni"
+	"github.com/tiffanyfay/aws-k8s-test-framework/test/e2e/framework"
+	"github.com/tiffanyfay/aws-k8s-test-framework/test/e2e/resources"
+
+	log "github.com/cihub/seelog"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	promv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Timeout for waiting events in seconds// const TIMEOUT = 60
+var (
+	err                    error
+	testerNodeName         string
+	testPodErrPercentLimit float64
+	awsNodeErrLimit        int64
+
+	f                *framework.Framework
+	prom             *resources.Prom
+	promResources    *resources.Resources
+	testpodResources *resources.Resources
+	resourcesGroup   []*resources.Resources
+
+	nodes   []corev1.Node
+	promAPI promv1.API
+)
+
+var _ = Describe("Testing CNI", func() {
+	f = framework.New()
+	ctx := context.Background() // TODO make this have a timeout
+
+	ns := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "cni-test"}}
+	kubeSystem := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "kube-system"}}
+	awsNodeDS := &appsv1.DaemonSet{ObjectMeta: metav1.ObjectMeta{Name: "aws-node", Namespace: kubeSystem.Name}}
+
+	// testPodErrPercentLimit := 0.1
+	awsNodeErrLimit = 5
+
+	Context("Testing with 1 node per test", func() {
+		It("Should wait for desired number of nodes", func() {
+			desired := 3
+
+			nodeList, err := f.ClientSet.CoreV1().Nodes().List(metav1.ListOptions{})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			err = cni.WaitForASGDesiredCapacity(ctx, f, nodeList.Items, desired)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			testerNodeName, err = cni.GetTesterPodNodeName(f, ns.Name, "cni-e2e")
+			Expect(err).ShouldNot(HaveOccurred())
+
+			nodes, err = cni.GetTestNodes(f, testerNodeName)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(len(nodes)).To(BeNumerically(">=", desired-1))
+		})
+		Context("Run tests", func() {
+			BeforeEach(func() {
+				By("Creating prometheus resources")
+				promResources = resources.NewPromResources(ns.Name, testerNodeName, 1)
+				promResources.ExpectDeploySuccessful(ctx, f, ns)
+
+				promAPI, err = resources.NewPromAPI(f, ns)
+				Expect(err).NotTo(HaveOccurred())
+				time.Sleep(time.Second * 5)
+
+				prom = &resources.Prom{API: promAPI}
+				time.Sleep(time.Second * 5)
+
+				resourcesGroup = createTestResources(ctx, ns, nodes)
+			})
+
+			It("Should pass with WARM_IP_TARGET=0 (default), WARM_ENI_TARGET=1 (default), and MAX_ENI=-1(default)", func() {
+				cni.UpdateAWSNodeEnvs(ctx, f, "0", "1", "-1")
+				i := 0
+				internalIP, _, ipLimit := setup(ctx, f, awsNodeDS, i)
+
+				By(fmt.Sprintf("scaling deployment (%s) to get %d pods and 3 ENIs", resourcesGroup[i].Deployment.Name, ipLimit*2))
+				resourcesGroup[i].ExpectDeploymentScaleSuccessful(ctx, f, ns, int32(ipLimit*2))
```

Comment:
For looking at the test or when it runs? Because when it runs it tells the expected #s. Not sure how to make it clear via a helper function if it's solely for the code.